gobblin-runtime/src/main/java/gobblin/runtime/job_monitor/KafkaJobMonitor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.runtime.job_monitor;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collection;
+
+import com.google.common.base.Charsets;
+import com.typesafe.config.Config;
+
+import gobblin.runtime.api.JobSpec;
+import gobblin.runtime.api.JobSpecMonitor;
+import gobblin.runtime.api.MutableJobCatalog;
+import gobblin.runtime.kafka.HighLevelConsumer;
+import gobblin.util.ConfigUtils;
+import gobblin.util.Either;
+
+import kafka.message.MessageAndMetadata;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * Abstract {@link JobSpecMonitor} that reads {@link JobSpec}s from a Kafka stream. Subclasses should implement
+ * {@link KafkaJobMonitor#parseJobSpec(byte[])} to transform the message into one or multiple {@link JobSpec}s.
+ */
+@Slf4j
+public abstract class KafkaJobMonitor extends HighLevelConsumer<byte[], byte[]> implements JobSpecMonitor {
+
+  public static final String KAFKA_JOB_MONITOR_PREFIX = "jobSpecMonitor.kafka";
+
+  private final MutableJobCatalog jobCatalog;
+
+  /**
+   * @return A collection of either {@link JobSpec}s to add/update or {@link URI}s to remove from the catalog,
+   *        parsed from the Kafka message.
+   * @throws IOException
+   */
+  public abstract Collection<Either<JobSpec, URI>> parseJobSpec(byte[] message) throws IOException;
```

Comment:
We could run multiple jobs from a trigger. For example, when receiving a data availability event we may want to launch copies to multiple clusters, as well as a format transformation, etc.